libs/text-splitters/langchain_text_splitters/base.py

```diff
@@ -36,6 +36,7 @@ def __init__(
         chunk_overlap: int = 200,
         length_function: Callable[[str], int] = len,
         keep_separator: bool = False,
+        separator_kept_at_end: bool = False,
```

Comment:
instead of adding another parameter thoughts on expanding the type of `keep_separator` to something like: `Union[bool, Literal["beginning", "end"]]`

Reply:
Did so! Let me know if this looks okay :)

Reply:
nice! could we add a simple unit test?

Reply:
Sure! Do you want only for the scenario where `keep_separator=="end"`?

Reply:
one for "start" and one for "end" would be great

Reply:
@baskaryan Done, please let me know if there is anything else to be done!

Reply:
@baskaryan Following up to see if we could close this?